src/main/java/dev/enginecode/eccommons/infrastructure/json/repository/JsonMappingRepository.java

```diff
@@ -1,82 +1,57 @@
 package dev.enginecode.eccommons.infrastructure.json.repository;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.querydsl.core.NonUniqueResultException;
-import com.querydsl.sql.Configuration;
-import com.querydsl.sql.PostgreSQLTemplates;
 import com.querydsl.sql.SQLQueryFactory;
-import com.querydsl.sql.SQLTemplates;
-import dev.enginecode.eccommons.exception.JsonNotDeserializedException;
 import dev.enginecode.eccommons.exception.ResourceNotFoundException;
 import dev.enginecode.eccommons.exception.TableNotFoundException;
 import dev.enginecode.eccommons.infrastructure.json.model.TableAnnotatedRecord;
 import dev.enginecode.eccommons.infrastructure.json.model.TableName;
+import dev.enginecode.eccommons.infrastructure.json.repository.database.DatabaseConnection;
+import dev.enginecode.eccommons.infrastructure.json.repository.database.PostgresDatabaseConnection;
+import dev.enginecode.eccommons.infrastructure.json.repository.mapping.JSONMapper;
+import dev.enginecode.eccommons.infrastructure.json.repository.mapping.Mapping;
 import org.springframework.stereotype.Repository;
 
 import javax.sql.DataSource;
 import java.io.Serializable;
+import java.util.Optional;
 
-import static dev.enginecode.eccommons.infrastructure.json.errors.InfrastructureErrorCode.*;
+import static dev.enginecode.eccommons.infrastructure.json.errors.InfrastructureErrorCode.RESOURCE_NOT_FOUND;
+import static dev.enginecode.eccommons.infrastructure.json.errors.InfrastructureErrorCode.TABLENAME_ANNOTATION_EMPTY;
 
 @Repository
 public class JsonMappingRepository<ID extends Serializable> implements JsonRepository<ID> {
```

Comment:
Now, following the naming convention `PostgresDatabaseConnection` introduced, I think that this class could also have name specific for the particular database system like `JsonPostgresRepository`, what do you think?

Reply:
OK, renamed.